genexuslabs/web-controls-library
# Add line clamping for gx-textblock and gx-edit

#### Changes proposed in this Pull Request

- Added support for line clamping for `gx-textblock` and `gx-edit`. To enable line clamping, set `lineClamp` property to true.

The mechanism currently works only in Webkit based browsers, through the `-webkit-line-clamp` property. However, other steps could be taken to add support for other browsers in the future, by following an approach similar to `shave` library.

#### Testing instructions

Add the following HTML snippet:

```
    <gx-table style="width: 200px; height: 60px">
      <gx-table-cell overflow-mode="clip" line-clamp
        ><gx-textblock line-clamp
          >Some long text here Some long text here Some long text here Some long
          text here Some long text here Some long text here Some long text here
          Some long text here Some long text here Some long text here Some long
          text here Some long text here Some long text here Some long text here
          Some long text here Some long text here Some long text here Some long
          text here Some long text here Some long text here Some long text here
          Some long text here Some long text here Some long text here Some long
          text here
        </gx-textblock></gx-table-cell
      >
    </gx-table>

    <gx-table style="width: 200px; height: 60px">
      <gx-table-cell overflow-mode="clip" line-clamp
        ><gx-edit
          line-clamp
          value="Some long text here Some long text here Some long text here Some long
          text here Some long text here Some long text here Some long text here
          Some long text here Some long text here Some long text here Some long
          text here Some long text here Some long text here Some long text here
          Some long text here Some long text here Some long text here Some long
          text here Some long text here Some long text here Some long text here
          Some long text here Some long text here Some long text here Some long
          text here"
          readonly
        >
        </gx-edit
      ></gx-table-cell>
    </gx-table>
```

Verify that for both `gx-edit` and `gx-textblock` an ellipsis is rendered and the text is nicely truncated.
